service/payment.go

```diff
@@ -121,45 +122,90 @@ func (service *PaymentService) GetPaymentSession(w http.ResponseWriter, req *htt
 		return
 	}
 
-	paymentResource, err := service.DAO.GetPaymentResource(id)
-	if paymentResource == nil {
-		log.Info(fmt.Sprintf("payment session not found. id: %s", id))
-		w.WriteHeader(http.StatusForbidden)
+	paymentSession, httpStatus, err := (*PaymentService).getPaymentSession(service, id)
+	if err != nil {
+		w.WriteHeader(httpStatus)
+		log.ErrorR(req, err)
 		return
 	}
+
+	w.Header().Set("Content-Type", "application/json")
```

Comment:
Just a thought I could be wrong, but can we get this header from the request.

Reply:
According to the spec, this is correct.